smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -2301,8 +2301,12 @@ private HttpBinding readPayload(
         // Handle streaming shapes differently.
         if (target.hasTrait(StreamingTrait.class)) {
             if (target instanceof UnionShape) {
-                // If payload is a event stream, return it after calling event stream deser function.
-                generateEventStreamDeserializer(context, binding.getMember(), target);
+                // If payload is an event stream, return it after calling event stream deser function.
+                HttpProtocolGeneratorUtils.generateHttpBindingEventStreamDeserializer(
+                        context, target, "const data: any",
+                        deserializingEventShapes,
+                        deserializeEventUnions
+                );
```

Comment:
nit: the formating here seems a bit off. 

Also nit: since `deserializingEventShapes` and `deserializeEventUnions` are internal data to the `HttpBindingProtocolGenerator` class, I'm not a big fan of sharing them with other classes. How about we move the logic of adding target shape to these data structures out of this function? And this function will only generate the code. For example 

```suggestion
                // If payload is an event stream, return it after calling event stream deser function.
                HttpProtocolGeneratorUtils.generateHttpBindingEventStreamDeserializer(
                        context, target, "const data: any"
                );
                if (target instanceof StructureShape) {
                    // Single-event stream. Save the structure and generate event-specific deser later.
                     deserializingEventShapes.add(target.asStructureShape().get());
                } else if (target instanceof UnionShape) {
                    // Multi-event stream. Save the union and generate dispatcher later.
                    deserializeEventUnions.add(target.asUnionShape().get());
                } else {
                    throw ...
                }
                
```